lib/src/path_parser.cc

```diff
@@ -143,14 +143,18 @@ namespace hocon {
             return token_list { t };
         }
 
+        // The split iterator tacks an empty string onto the end of the iterator if the string
+        // ends in the split character; so we need to remove trailing '.'
+        string text_to_split = boost::trim_right_copy_if(token_text, is_any_of("."));
```

Comment:
Line 165 seems like it duplicates this. Why didn't it work? (also it could've used `token_text.back()`).